test/unit/manipulation.js

```diff
@@ -1620,6 +1620,8 @@ function testHtml( valueObj ) {
 	tmp = " " + tmp.slice( 1 );
 	equal( div.html(valueObj( tmp )).html().replace( />/g, "&gt;" ), tmp, "Escaped html, leading space" );
 
+	equal( jQuery("<tr-d></tr-d><tr-d></tr-d>").length, 2, "Create tag-hyphenated elements." );
```

Comment:
I would want more assertions for this, so it would be more thorough, like check for nodeName and for other elements like <a href="https://github.com/jquery/jquery/blob/740e190223d19a114d5373758127285d14d6b71e/src/manipulation.js#L31-47">this</a>. No need to do this for core module though.

And i would prefer if those assertions would be in different test.